provider/datagen/src/driver.rs

```diff
@@ -462,17 +438,34 @@ impl DatagenDriver {
     /// [`langid!`]: crate::prelude::langid
     /// [`DatagenProvider::locales_for_coverage_levels`]: crate::DatagenProvider::locales_for_coverage_levels
     pub fn with_locales_and_fallback(
-        self,
+        mut self,
         locales: impl IntoIterator<Item = LocaleFamily>,
         options: FallbackOptions,
     ) -> Self {
-        Self {
-            locales_fallback: Some(LocalesWithOrWithoutFallback::WithFallback {
-                families: locales.into_iter().map(LocaleFamily::into_parts).collect(),
-                options,
-            }),
-            ..self
+        let mut families = locales.into_iter().collect::<Vec<_>>();
+        if families.is_empty() {
```

Comment:
if we remove this case, the `no_fallback` method can call this one. Do we still need this in 2.0?

Reply:
I'm ok fixing this